### DISCUSSION TITLE
MediaElement with Bearer authentication

### DISCUSSION THREAD

**Discussion by Pmr-precure** [CommunityToolkit/Maui#2000] [General]
Hallo,

Is there a way of streaming a video in the MediaElement from a protected API that returns a FileStreamResult.

The thing is we got an API that return some videos. And we want to use the MediaElement to show them. Its easy enough providing the URL when authentication is off. But when its on we need a way of passing the JWT.

Hope you can help :)

BR.

**Comment by ghost** [comment]
Just as an idea talking with CoPilot, you could use httpRequest with Bearer authentication, download the video and put the stream (byte array) to MediaElement.
